content/doc/developer/extensibility/index.adoc

```diff
@@ -22,4 +22,30 @@ references:
   title: My class is missing descriptor
 - url: https://wiki.jenkins-ci.org/display/JENKINS/Extension+Point+for+Project+Views+Navigation
   title: Extension Point for Project Views Navigation # TODO make into guide or remove as obsolete with BO
+- url: https://github.com/jenkinsci/sample-extensions
+  title: Sample extension implementations
 ---
+
+Jenkins is designed with extensibility in mind, both internally (core) and with plugins.
+Extensibility in Jenkins is accomplished by combining the concepts discussed below.
+
+== Extension annotation
+
++jenkinsdoc:Extension[]+ is an annotation that allows Jenkins to discover classes implementing certain types, instantiate them, and register them in global lists of implementations of its supertypes and interfaces.
```

Comment:
This sentence is a little confusing to me at "register them in global lists of implementations of its supertypes and interfaces."

I think what this sentence is saying is something along the lines of: Extension is an annotation that allows Jenkins to discover classes implementing certain interfaces [an extension is only going to be an interface right? the use of "type" in the original sentence is confusing to me too], and instantiate objects from those classes. To accomplish this, Jenkins maintains global lists, based on the extension type, of annotated classes.


My phrasing might also not be good, but I tried to just parrot what I understood from the original.